src/query/list_file.c

```diff
@@ -5686,64 +5542,44 @@ qfile_lookup_list_cache_entry (THREAD_ENTRY * thread_p, int list_ht_no, const DB
   /* look up the hash table with the key */
   lent = (QFILE_LIST_CACHE_ENTRY *) mht_get (qfile_List_cache.list_hts[list_ht_no], params);
   qfile_List_cache.lookup_counter++;	/* counter */
+
   if (lent)
     {
+      unsigned int i;
+
+#if defined(SERVER_MODE)
       /* check if it is marked to be deleted */
       if (lent->deletion_marker)
 	{
-	  (void) qfile_delete_list_cache_entry (thread_p, lent);
-	  lent = NULL;
-	}
-#if defined(SERVER_MODE)
-#if 0
-      /* check if the recorded transaction isolation level is higher than the current one */
-      if (lent)
-	{
-	  tran_isolation = logtb_find_isolation (tran_index);
-	  if (lent->tran_isolation < tran_isolation)
-	    {
-	      lent = NULL;
-	    }
+	  qfile_delete_list_cache_entry (thread_p, lent);
+	  goto end;
 	}
-#else /* 1 */
-      /* check if it may have uncommitted data */
-      if (lent)
+
+      *result_cached = true;
+
+      /* check if the cache is owned by me */
+      for (i = 0; i < lent->last_ta_idx; i++)
 	{
-	  tran_isolation = logtb_find_isolation (tran_index);
-	  /* 1. my isolation is not (greater than) read uncommited 2. the entry is cached by an uncommitted other
-	   * transaction */
-	  num_elements = (int) lent->last_ta_idx;
-	  if (lent->uncommitted_marker == true)
+	  if (lent->tran_index_array[i] == tran_index)
 	    {
-	      /* treat as look-up failed,
-	       * because the cache is assigned already by other transaction */
-	      assert (lent->last_ta_idx > 0);
-	      if (lent->tran_index_array[lent->last_ta_idx - 1] != tran_index)
-		{
-		  lent->last_ta_idx = num_elements;
-		  lent = NULL;
-		}
-	    }
-	  else
-	    {
-	      lent->last_ta_idx = num_elements;
+	      new_tran = false;
+	      break;
 	    }
 	}
-#endif /* 1 */
-#endif /* SERVER_MODE */
+#endif
 
+#if defined(SERVER_MODE)
       /* finally, we found an useful cache entry to reuse */
-      if (lent)
+      if (new_tran)
 	{
 	  /* record my transaction id into the entry and adjust timestamp and reference counter */
-#if defined(SERVER_MODE)
+	  lent->uncommitted_marker = true;
```

Comment:
The logic related to uncommitted_marker has been removed, but it is still used.

Reply:
uncommitted_marker is used at printing the status of result-cache.
for example, at csql
```
csql> ;info qcache
....
uncommitted_marker = true
...
```

Reply:
I recommend deleting the uncommitted_marker. uncommitted_marker is meaningless.

Reply:
I removed it from result-cache dump (;info qcache)